test/functional/api/v2/media_controller_test.rb

```diff
@@ -26,7 +26,7 @@ class Api::V2::MediaControllerTest < ActionController::TestCase
     assert_difference('Medium.count', +1) do
       post :create, { :medium => new_medium }
     end
-    assert_response :created
+    assert_response :success
```

Comment:
All the rest of the tests are(except those two) check for success status being returned and that is why i changed this two, but you are right it should return a 201.
The problem is that once it has a template it goes to the rabl template straight.
I can add something like that in the base controller : 

      render_status = self.env["REQUEST_METHOD"] == "POST" ? 201 : 200
      response ||= get_resource
      respond_with response, :responder => ApiResponder, :status => render_status

you want to do that?